### ISSUE TITLE
False positive for RS1024 when creating hash-based collections

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn-analyzers#4845]
### Analyzer

**Diagnostic ID**: [RS1024](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/rs1024): `Compare symbols correctly`

### Packages

- `Microsoft.CodeAnalysis.CSharp` version 3.8.0
- `Microsoft.CodeAnalysis.Analyzers` version 3.3.2
- Visual Studio 16.9 Preview 4

### Describe the bug

The RS1024 warning is displayed when creating hash-based collections with an `ISymbol` as key, even though the correct symbol comparer is passed to the collection constructors.

### Steps To Reproduce

Paste this snippet:

```csharp
HashSet<INamedTypeSymbol> discoveredTypes = new(SymbolEqualityComparer.Default);
Dictionary<IMethodSymbol, MethodDeclarationSyntax> staticMethods = new(SymbolEqualityComparer.Default);
Dictionary<IFieldSymbol, string> constantDefinitions = new(SymbolEqualityComparer.Default);
```

### Expected behavior
No warnings.

### Actual behavior
RS1024 warnings on all three lines.

## Screenshot
![image](https://user-images.githubusercontent.com/10199417/107826920-10cf5300-6d86-11eb-894f-ca6abe8064fb.png)

**Comment by Youssef1313** [comment]
Looks a duplicate of https://github.com/dotnet/roslyn-analyzers/issues/4568 which was fixed by @Evangelink.

@Evangelink Can you confirm please?

**Comment by Youssef1313** [comment]
@Sergio0694 Meanwhile, can you try [3.3.3-beta1.21109.6](https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet6&package=Microsoft.CodeAnalysis.Analyzers&version=3.3.3-beta1.21109.6&protocolType=NuGet)?

**Comment by Sergio0694** [comment]
I can confirm that using the `3.3.3-beta1.21109.6` version from the Azure feed fixes the issue, thanks! 😄
Closing this then as it was indeed already addressed by #4571.
